packages/data-stores/src/woocommerce-eligibility/reducer.ts

```diff
@@ -0,0 +1,44 @@
+import { combineReducers } from '@wordpress/data';
+import { TransferEligibility } from './types';
+import type { Action } from './actions';
+import type { Reducer } from 'redux';
+
+export const transferEligibility: Reducer<
+	{ [ key: number ]: TransferEligibility } | undefined,
+	Action
+> = ( state, action ) => {
+	switch ( action.type ) {
+		case 'TRANSFER_ELIGIBILITY_RECEIVE': {
+			return {
+				...state,
+				[ action.siteId ]: action.transferEligibility,
+			};
+		}
+	}
+
+	return state;
+};
+
+// Tracks product list fetching state
+export const isFetchingTransferEligibility: Reducer< boolean | undefined, Action > = (
```

Comment:
The type is `Reducer< boolean >`. The reducer state is never `undefined`. Except for a short time during initialization, but that's invisible to the user.